### PR TITLE
Implement task status reporting in providers (closes #1320)

### LINKED ISSUE #1320: SauceLabs integration: report test status (success/fail) to the SauceLabs service
### Are you requesting a feature or reporting a bug?

A feature

### What is the current behavior?

After test job has been completed, SauceLabs task status is "unknown".
"This status is set if one or more jobs of the latest build haven't been reported as passed or failed. This state is equivalent to Finished."

### What is the expected behavior?

Test results status is reported to the SauceLabs service.

#### Provide the test code and the tested page URL (if applicable)

https://github.com/surveyjs/surveyjs

Take a look to the sauce labs test status badge.

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

File: src/runner/index.js [hunk 1]
```diff
@@ -67,7 +68,10 @@ export default class Runner extends EventEmitter {
 
     // Run task
     async _getTaskResult (task, browserSet, reporter, testedApp) {
-        task.on('browser-job-done', job => browserSet.releaseConnection(job.browserConnection));
+        task.on('browser-job-done', job => {
+            job.browserConnection.reportTaskStatus(TASK_STATUS.done, { total: job.total, passed: job.passed });
```
Comment (inikulin) [inline src/runner/index.js]: In this context it's not task status, but browser job status. Browser job can be  done at this point, but whole task is still running. Therefore, maybe it makes sense to rename it to `reportBrowserJobStatus` throughout the codebase?

File: src/runner/index.js [hunk 2]
```diff
@@ -81,6 +85,7 @@ export default class Runner extends EventEmitter {
             await Promise.race(promises);
         }
         catch (err) {
+            await browserSet.reportTaskStatus(TASK_STATUS.errored, err);
```
Comment (inikulin) [inline src/runner/index.js]: Here we report error for all browsers, however as far as I understand it makes sense to report errors only in this particular browser.

Comment (inikulin) [inline src/runner/index.js]: Taking in the account all the remarks above,  guess it's makes sense to move this functionality to the `BrowserJob` class

File: src/runner/task-status.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export default {
+    done:     'task-done',
```
Comment (inikulin) [inline src/runner/task-status.js]: If we call it `status` then where should be a value for all phases of the browser job. E.g. `running` and `pending` are missing. However, if we'll rename it to "browser job result" these values will be legit.

**Comment by AndreyBelym** [comment]
FPR

**Comment by AndreyBelym** [comment]
\cc @inikulin @AlexanderMoskovkin

File: src/runner/browser-job.js [hunk 4]
```diff
@@ -15,15 +16,33 @@ export default class BrowserJob extends EventEmitter {
...
+        this.connectionErrorListener = error => this._setResult(RESULT.errored, error);
+
+        this.browserConnection.once('error', this.connectionErrorListener);
+    }
+
+    async _setResult (code, data) {
+        if (this.result)
+            return;
+
+        this.result = { code, data };
```
Comment (AlexanderMoskovkin) [inline src/runner/browser-job.js]: I suppose it should be `status` instead of `code`

**Comment by AndreyBelym** [comment]
FPR

File: src/runner/browser-job.js [hunk 5]
```diff
@@ -171,4 +198,10 @@ export default class BrowserJob extends EventEmitter {
 
         return null;
     }
+
+    cancel () {
```
Comment (inikulin) [inline src/runner/browser-job.js]: Why not `abort` for consistency?

**Comment by AndreyBelym** [comment]
FPR

**Pull request merged**